src/Analyzers/Core/CodeFixes/MakeFieldReadonly/AbstractMakeFieldReadonlyCodeFixProvider.cs

```diff
@@ -67,7 +67,9 @@ private async Task MakeFieldReadonlyAsync(Document document, SyntaxEditor editor
 
                 if (declarationDeclarators.Count == fieldDeclarators.Count())
                 {
-                    editor.SetModifiers(fieldDeclarators.Key, editor.Generator.GetModifiers(fieldDeclarators.Key) | DeclarationModifiers.ReadOnly);
+                    var modifiers = editor.Generator.GetModifiers(fieldDeclarators.Key) - DeclarationModifiers.Volatile | DeclarationModifiers.ReadOnly;
```

Comment:
1. using `-` here is very uncommon.  Generally to wipe the bits here it's preferred to do `x & ~y`.  
2. this pattern is repeated twice.  can you extract a helper for this?

Reply:
This isn't an Enum. ```DeclarationModifiers``` doesn't have override for ```~``` operator.